python/iceberg/api/expressions/literals.py

```diff
@@ -76,6 +77,7 @@ class Literal(object):
     JAVA_MIN_INT = -2147483648
     JAVA_MAX_FLOAT = 3.4028235E38
     JAVA_MIN_FLOAT = -3.4028235E38
+    value: Any
```

Comment:
Not sure I love the class variable here. And its not being used below (the `self.value` in `BaseLiteral` hides it). Maybe thsi should be an `abc.ABC`?

Reply:
Let me take a look, that sounds reasonable

Reply:
Im still musing about adjusting the class heirarchy. I think we can skip the `abc.ABC` for now. I do think the class variable can go though.

Reply:
I believe the `value` is there purely to satisfy mypy checks(per [this PEP](https://www.python.org/dev/peps/pep-0526/) .  I do think you are correct that the Literal Class is kinda useless, I'll need to double check, but I think I could just make everything BaseLiteral, which does already have `value` as a instance variable, and move the constants and the `of` method  out to be in the file global scope.

Reply:
thats cool. I think we can also just remove the `value: Any` since its not used anywhere else in the class. The static method doesn't use it right?

Reply:
yeah, it's not used anywhere it's just a type hint.

Reply:
I removed `value` from `Literal` and changed the type hints in `Predicate` to `BaseLiteral`. Let me know if that looks good now.